python_modules/libraries/dagster-airflow/dagster_airflow/resources/airflow_ephemeral_db.py

```diff
@@ -0,0 +1,172 @@
+import datetime
+import importlib
+import os
+import tempfile
+from typing import Generator, List, Mapping, Optional
+
+import airflow
+import pytz
+from airflow.models.connection import Connection
+from airflow.models.dag import DAG
+from airflow.models.dagrun import DagRun
+from airflow.utils import db
+from dagster import (
+    Array,
+    DagsterInvariantViolationError,
+    Field,
+    InitResourceContext,
+    ResourceDefinition,
+    _check as check,
+)
+from dagster._core.instance import AIRFLOW_EXECUTION_DATE_STR
+from dateutil.parser import parse
+
+from dagster_airflow.utils import (
+    Locker,
+    create_airflow_connections,
+    is_airflow_2_loaded_in_environment,
+    serialize_connections,
+)
+
+# pylint: disable=no-name-in-module,import-error
+if is_airflow_2_loaded_in_environment():
+    from airflow.utils.state import DagRunState
+    from airflow.utils.types import DagRunType
+else:
+    from airflow.utils.state import State
+
+    # pylint: enable=no-name-in-module,import-error
+
+
+class AirflowEphemeralDatabase:
+    """
+    A ephemeral Airflow database Dagster resource.
+
+    """
+
+    def __init__(self, context: InitResourceContext):
+        self.airflow_home_path = os.path.join(
+            tempfile.gettempdir(), f"dagster_airflow_{context.run_id}"
+        )
+        self.context = context
+        os.environ["AIRFLOW_HOME"] = self.airflow_home_path
+        os.makedirs(self.airflow_home_path, exist_ok=True)
+        with Locker(self.airflow_home_path):
+            airflow_initialized = os.path.exists(f"{self.airflow_home_path}/airflow.db")
+            # because AIRFLOW_HOME has been overriden airflow needs to be reloaded
+            if is_airflow_2_loaded_in_environment():
+                importlib.reload(airflow.configuration)
+                importlib.reload(airflow.settings)
+                importlib.reload(airflow)
+            else:
+                importlib.reload(airflow)
+            if not airflow_initialized:
+                db.initdb()
+                create_airflow_connections(
+                    [Connection(**c) for c in context.resource_config["connections"]]
+                )
```

Comment:
recommend not doing anything this stateful and heavy in an `__init__` method. It makes classes much more difficult to test and reuse in other contexts. I would instead have a factory function/static method that takes the context, does the super heavy global state stuff, and gets only the data it needs out of the context.